packages/ra-core/src/controller/useListParams.ts

```diff
@@ -156,8 +156,11 @@ const useListParams = ({
     }, requestSignature); // eslint-disable-line react-hooks/exhaustive-deps
 
     const setSort = useCallback(
-        (newSort: string) =>
-            changeParams({ type: SET_SORT, payload: { sort: newSort } }),
+        (newSort: string, newOrder?: string) =>
```

Comment:
I'd name the params `sort` and `order`, it would make the changeParams payload as simple as `{ sort, order }`